pkg/operation/seed/scheduler/gardener_kube_scheduler.go

```diff
@@ -0,0 +1,395 @@
+						Scope:       &scope,
+						Resources:   []string{"pods"},
+					},
+				}},
+				FailurePolicy: &failPolicy,
+				MatchPolicy:   &matchPolicy,
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot,
+					},
+				},
+				ObjectSelector:          &metav1.LabelSelector{},
+				SideEffects:             &sideEffects,
+				TimeoutSeconds:          &timeout,
+				AdmissionReviewVersions: []string{admissionregistrationv1beta1.SchemeGroupVersion.Version},
+				ReinvocationPolicy:      &revocationPolicy,
+			}},
+		}
+		vpa = &autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: k.namespace,
+				Labels:    getLabels(),
+			},
+			Spec: autoscalingv1beta2.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+					Name:       deployment.Name,
+				},
+				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
+					UpdateMode: &updateMode,
+				},
+			},
+		}
+
+		registry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+	)
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, namespace, func() error {
+		namespace.Labels = utils.MergeStringMaps(namespace.Labels, getLabels())
+		return nil
+	}); err != nil {
+		return errors.Wrap(err, "update of Namespace failed")
+	}
+
+	resources, err := registry.AddAllAndSerialize(
+		kubeSchedulerClusterRoleBinding,
+		volumeSchedulerClusterRoleBinding,
+		roleBinding,
+		serviceAccount,
+		configMap,
+		deployment,
+		webhook,
+		vpa,
+	)
+	if err != nil {
+		return err
+	}
+
+	return common.DeployManagedResourceForSeed(ctx, k.client, "gardener-kube-scheduler", k.namespace, false, resources)
+}
+
+func getLabels() map[string]string {
+	return map[string]string{
+		"app":  "kubernetes",
+		"role": "scheduler",
+	}
+}
+
+func (k *kubeScheduler) Destroy(ctx context.Context) error {
+	return client.IgnoreNotFound(k.client.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: k.namespace}}))
+}
+
+func (k *kubeScheduler) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Minute*2)
+	defer cancel()
+
+	return managedresources.WaitUntilManagedResourceHealthy(timeoutCtx, k.client, k.namespace, "gardener-kube-scheduler")
+}
+
+func (k *kubeScheduler) WaitCleanup(ctx context.Context) error {
+	return retry.UntilTimeout(ctx, time.Second*2, time.Minute*2, func(ctx context.Context) (bool, error) {
+		if err := k.client.Get(ctx, client.ObjectKey{Name: k.namespace}, &corev1.Namespace{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				return retry.Ok()
+			}
+			return retry.MinorError(err)
+		}
+		return retry.MinorError(errors.Errorf("Namespace %s is not deleted yet", k.namespace))
```

Comment:
You can use https://github.com/gardener/gardener/blob/4c6153dceae2e461607c9a38afca88277387dda0/pkg/utils/kubernetes/kubernetes.go#L128 instead :)